src/main/java/com/eva/logic/commands/EditCommand.java

```diff
@@ -100,9 +121,30 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
-        Set<Comment> updatedComments = editPersonDescriptor.getComments().orElse(personToEdit.getComments());
+        Set<Comment> editedComments = editPersonDescriptor.getComments();
+        Set<Comment> currentComments = personToEdit.getComments();
+        if (editedComments != null) {
+            for (Comment comment: currentComments) {
+                for (Comment editedComment : editedComments) {
+                    if (editedComment.getTitle().equals(comment.getTitle())
+                            && editedComment.getDate().equals(comment.getDate())) {
+                        comment.update(editedComment.getDescription());
+                    }
+                }
```

Comment:
Maybe you can abstract this out into another function declared in this file. Something like 
public void updateComments(Set<Comment> currentComments, Set<Comment> editedComments)